### PR TITLE
DBM Stage Trigger: new "stage total" option

### PR DESCRIPTION
**Pull request by mrbuds**
this option is supported since
https://github.com/DeadlyBossMods/DBM-Unified/commit/407e3ba5156409b5fe6becce35d0662014bb2d1a

Renamed previous "Stage" option to "Current Stage"
Called new option "Stage Total", i don't know if "Total Stage" is better or not.

### CODE REVIEW

File: WeakAuras/Prototypes.lua [hunk 1]
```diff
@@ -4889,11 +4889,25 @@ Private.event_prototypes = {
...
+        display = L["Current Stage"],
         type = "number",
         conditionType = "number",
         store = true,
-      }
+      },
+      {
+        name = "stageTotal",
+        init = "select(2, WeakAuras.GetDBMStage())",
+        display = L["Stage Total"],
```
[the remark this answers is not in the document]
Comment (mrbuds) [reply WeakAuras/Prototypes.lua:4900]: That's it

**Comment by MysticalOS** [comment]
The gist of it is, up until now SetStage callback and internal phase tracking of DBM has always matched journal.

So for example eye of the jailer is 1, then 2, then 1 again, then 2 again, then 3.
Kel thuzad is also same way, 1, 2, 1, 2 until final push then phase 3. Exactly like journal
Other examples of matching journal would be sylvanas. In journal she's phase 1, intermission, 2, 3.
So DBM did her phases as 1, 1.5, 2, 3. Typically intermissions are point 5s because i still want the whole phases to match journal.

The second scenario didn't really cause issues with weak auras. it didn't match bigwigs most of time which typically numbers stages as whole numbers regardless of what journal says, but it was still workable. The first issue however made it not possible to setup special considerations for SPECIFIC times you returned to a repeat phase (ie second time you go into stage 1 on jailer only, versus always triggering).

Thus the phase totality count. this means someone can now make a weak aura that only triggers on SECOND phase 1 by using totality phase 4. if that makes sense?

I still think there is great merit in continuing to have first return of set stage return fights current journal phase, but now the second return gives niche weak auras total control over triggering on specific repeat phases on bosses like kelthuzad and eye of jailer specifically.  it also gives them a diff way to handle fights like sylvanas too if they want since it'll also more typically match way Bigiwigs reports it.

**Comment by InfusOnWoW** [comment]
Yeah I understand the use case and we should add it. But "Totality" or "Total" doesn't explain what the it is.

We need either a better word, or a tooltip. And probably both.

I guess for the "current stage", the tooltip could be: "Matches phase number of encounter journal, intermissions are .5". This implies that for cycled encounters, the numbers cycle too.

And for Stage Counter (?), the tooltip could be: "Increases by one per stage, intermission."

**Comment by mrbuds** [comment]
Updated names & tooltips

**Review by InfusOnWoW — APPROVED** [review]

**Pull request merged**